tests/lineup/internal/cmds.test.ts

```diff
@@ -0,0 +1,88 @@
+/// <reference types="jest" />
+import {serializeLineUpFilter, restoreRegExp} from '../../../src/lineup/internal/cmds';
+
+// The following tests worked with LineUp v3.
+// With LineUp v4 the filter object changed
+// describe('Serialize LineUp v3 filter for provenance graph', () => {
+//   it('filter as string', () => {
+//     expect(serializeRegExp('abc')).toBe('abc');
+//     expect(typeof serializeRegExp('abc')).toBe('string');
+//   });
+
+//   it('filter as RegExp', () => {
+//     expect(serializeRegExp(/abc/gm)).toMatchObject({value: '/abc/gm', isRegExp: true});
+//     expect(serializeRegExp(/abc/gm)).not.toMatchObject({value: '/12345/gm', isRegExp: true});
+//     expect(serializeRegExp(/abc/gm)).not.toMatchObject({value: '/abc/gm', isRegExp: false});
+//   });
+// });
+
+describe('Serialize LineUp v4 filter for provenance graph', () => {
+  it('filter as string', () => {
+    const lineUpFilter = {filter: 'abc', filterMissing: false};
+
+    expect(serializeLineUpFilter(lineUpFilter)).toMatchObject({filter: {value: 'abc', isRegExp: false}, filterMissing: false});
+  });
+
+  it('filter as RegExp', () => {
+    const lineUpFilter = {filter: /abc/gm, filterMissing: false};
+
+    expect(serializeLineUpFilter(lineUpFilter)).toMatchObject({filter: {value: '/abc/gm', isRegExp: true}, filterMissing: false});
+    expect(serializeLineUpFilter(lineUpFilter)).not.toMatchObject({filter: {value: '/12345/gm', isRegExp: true}, filterMissing: false});
+    expect(serializeLineUpFilter(lineUpFilter)).not.toMatchObject({filter: {value: '/abc/gm', isRegExp: false}, filterMissing: false});
+  });
+
+  it('filter as array', () => {
+    const lineUpFilter = {filter: ['chromosome', 'gender'], filterMissing: false};
+    expect(serializeLineUpFilter(lineUpFilter)).toMatchObject({filter: {value: ['chromosome', 'gender'], isRegExp: false}, filterMissing: false});
+  });
+
+  it('filter as null', () => {
+    const lineUpFilter = {filter: null, filterMissing: false};
+    expect(serializeLineUpFilter(lineUpFilter)).toMatchObject({filter: {value: null, isRegExp: false}, filterMissing: false});
+  });
+
+  it('filter as null and use regular expression true', () => {
+    const lineUpFilter = {filter: /null/, filterMissing: false};
+    expect(serializeLineUpFilter(lineUpFilter)).toMatchObject({filter: {value: /null/, isRegExp: true}, filterMissing: false});
```

Comment:
What should happen in this case ?
![serialize](https://user-images.githubusercontent.com/51322092/78022079-a6234d80-7354-11ea-99a8-6d75224f8028.png)

Reply:
After a second inspection i found out that the value  `/null/` is not possible since the change regex filter checkbox reacts only if a value is present.

I added another test for the case when you click the filter missing values and filter is empty.

![filter](https://user-images.githubusercontent.com/51322092/78110970-38326100-73fc-11ea-9d09-81af5c518399.gif)

Reply:
Ok, good. Then we do not need to unit test this.